### PR TITLE
Output text for agents: raw TOML, markdown format, shared string utils

### PR DESCRIPTION
**Pull request by lchoquel**
## Summary

- **Raw TOML output** from `concept` and `pipe` commands (no JSON envelope)
- **Remove `assemble` command** — no consumers depend on it
- **`--format markdown|json`** option on `models` and `doctor` commands (default: markdown)
- **Shared `format_toml_string()` utility** extracted from MthdsFactory for consistent TOML string formatting in agent CLI

## Test plan

- [x] `make agent-check` passes (pyright, ruff, mypy, plxt)
- [x] `make agent-test` passes (all unit tests)
- [ ] Smoke test `pipelex-agent concept --spec '...'` outputs raw TOML
- [ ] Smoke test `pipelex-agent pipe --spec '...'` outputs raw TOML
- [ ] Smoke test `pipelex-agent models` outputs markdown
- [ ] Smoke test `pipelex-agent models --format json` outputs JSON
- [ ] Smoke test `pipelex-agent doctor --format json` outputs JSON
- [ ] Verify `pipelex-agent assemble` is no longer recognized

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Switch agent CLI outputs to raw TOML for `concept` and `pipe`, add markdown-first output to `models` and `doctor` with `--format markdown|json`, and remove `assemble`. Docs updated, markdown formatter tests added, and a skills handoff included to reflect the new output contract.

- **New Features**
  - `concept` and `pipe` now print raw TOML to stdout (no JSON envelope).
  - `models` and `doctor` support `--format markdown|json` (default: `markdown`) with human-friendly markdown output.
  - Shared `format_toml_string()` utility for consistent TOML strings.

- **Migration**
  - Update any consumers parsing JSON from `concept`/`pipe` to read raw TOML from stdout.
  - For previous JSON from `models`/`doctor`, pass `--format json`.
  - Remove any usage of `assemble`; no longer available.

<sup>Written for commit b9c4127f7e2e1d823948f6a22658d877338f2e72. Summary will update on new commits.</sup>

### CODE REVIEW

File: pipelex/cli/agent_cli/_agent_cli.py [hunk 1]
```diff
@@ -37,7 +36,6 @@ def list_commands(self, ctx: Context) -> list[str]:
             "inputs",
             "concept",
             "pipe",
-            "assemble",
             "models",
             "doctor",
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/cli/agent_cli/_agent_cli.py:40]: on purpose

File: pipelex/cli/agent_cli/commands/models_cmd.py [hunk 2]
```diff
@@ -233,11 +293,15 @@ def agent_models_cmd(
         str | None,
         typer.Option("--backend", "-b", help="Filter by backend name"),
     ] = None,
+    output_format: Annotated[
+        CliOutputFormat,
+        typer.Option("--format", help="Output format: markdown (default) or json (structured)"),
+    ] = CliOutputFormat.MARKDOWN,
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/cli/agent_cli/commands/models_cmd.py:299]: on purpose

File: pipelex/cli/agent_cli/commands/concept_cmd.py [hunk 3]
```diff
@@ -192,13 +192,7 @@ def concept_cmd(
...
         toml_content = _concept_spec_to_toml(concept_spec)
 
-        agent_success(
-            {
-                "success": True,
-                "concept_code": concept_spec.the_concept_code,
-                "toml": toml_content,
-            }
-        )
+        print(toml_content, end="" if toml_content.endswith("\n") else "\n")
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply pipelex/cli/agent_cli/commands/concept_cmd.py:195]: ➖ False positive — this is the intentional change of this PR. The `concept` and `pipe` commands now output raw TOML by design, replacing the JSON envelope. The docs (`docs/tools/cli/agent-cli.md` Output Contract section) have been updated to document the three output modes (JSON, raw TOML, markdown). The skills repo handoff note (`HANDOFF_SKILLS_OUTPUT_FORMAT.md`) covers the migration for downstream consumers.

File: tests/unit/pipelex/cli/test_format_models_markdown.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    def test_img_gen_category_present(self) -> None:
+        """Image Generation category with presets should appear."""
+        output = _format_models_markdown(self._FULL_RESULT)
+        assert "## Image Generation" in output
+        assert "- **gen-image**: Image generation" in output
+
+    def test_all_empty_produces_heading_only(self) -> None:
+        """Result with all empty sections should produce only the heading and hint."""
+        empty_result: dict[str, Any] = {
+            "presets": {"llm": [], "img_gen": [], "extract": [], "search": []},
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply tests/unit/pipelex/cli/test_format_models_markdown.py:102]: ➖ False positive — the test intentionally uses inline fixture data to test the markdown formatter in isolation. Importing `CATEGORY_DISPLAY_NAMES` from production code would make the test tautological (testing that the formatter iterates over the same dict it was given). The inline data also lets us test edge cases like missing categories without affecting other tests.

File: docs/features/cli.md [hunk 5]
```diff
@@ -51,11 +51,10 @@ The `pipelex-agent` CLI is a machine-first interface designed for automated envi
 | `fmt` | Format `.mthds`, `.toml`, or `.plx` files in-place |
 | `lint` | Lint files for errors |
 | `inputs` | Generate example input JSON for a pipe |
-| `concept` | Convert a JSON concept spec into TOML |
-| `pipe` | Convert a JSON pipe spec into TOML |
-| `assemble` | Combine TOML parts into a complete `.mthds` file |
-| `models` | List available model presets, aliases, and talent mappings |
-| `doctor` | Check configuration health and auto-fix issues |
+| `concept` | Convert a JSON concept spec into raw TOML (stdout) |
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply docs/features/cli.md:54]: ✅ Already fixed — the paragraph now reads: "Output format varies by command — JSON, raw TOML, or markdown — with no interactive prompts or Rich formatting. Errors always go to stderr as JSON." (commit b9c4127f)

File: docs/tools/cli/agent-cli.md [hunk 6]
```diff
@@ -87,17 +87,21 @@ These commands do not have subcommands:
...
-| `doctor` | Check config, credentials, and model health |
+| `concept` | Convert a JSON concept spec into raw TOML (stdout) |
+| `pipe` | Convert a JSON pipe spec into raw TOML (stdout) |
+| `models` | List available model presets, aliases, waterfalls, and talent mappings (`--format markdown\|json`, default: markdown) |
+| `doctor` | Check config, credentials, and model health (`--format markdown\|json`, default: markdown) |
 
 ## Output Contract
 
-Every command (except `fmt` and `lint`) returns structured JSON.
+Commands use different stdout formats depending on their purpose:
```
[the remark this answers is not in the document]
Comment (lchoquel) [reply docs/tools/cli/agent-cli.md:97]: ✅ Already fixed — the intro now reads: "Output format varies by command — JSON, raw TOML, or markdown — with no Rich formatting or interactive prompts." The Output Contract section documents all three modes. (commit b9c4127f)

**Review by thomashebrard — APPROVED** [review]
